core/core-spacemacs-buffer.el

```diff
@@ -455,6 +476,39 @@ ADDITIONAL-WIDGETS: a function for inserting a widget under the frame."
   "Insert quickhelp."
   (let ((widget-func
          (lambda ()
+           (when spacemacs-buffer-note-preview-lines
+             (let ((cursor-pos-before-widget-creation (point))
+                   (full-note-link-text
+                    (format "Click to show the full note (%s more lines)"
+                            spacemacs-buffer--note-preview-nr-of-removed-lines)))
+               ;; position cursor after the preview, inside the note frame
+               (forward-line -3)
+               (forward-char (- (/ (window-width) 2)
+                                (/ (length full-note-link-text) 2)))
+               ;; the 2 below is required to keep the note frame sides aligned,
+               ;; because the widget adds square brackets around the text
+               (delete-region (point) (+ (point) (length full-note-link-text) 2))
+               (add-to-list
+                'spacemacs-buffer--note-widgets
+                (widget-create
+                 'push-button
+                 :tag (propertize
+                       full-note-link-text 'face 'font-lock-warning-face)
+                 :help-echo "Open full note."
+                 :action (lambda (&rest ignore)
+                           (let ((cursor-pos-before-showing-full-note (point)))
```

Comment:
the let bindings can be simplified by using prog1

So instead of
(let ((var expr))
  body)
you can
(let ((var (progn1 expr body))))

prog1 runs both expr and body but the return value is expr.


Then you can combine the two let forms.

Reply:
I'm not sure what the prog1's first expression should return.

So I combined the two let bindings, since it doesn't matter what the note preview variable is set to when the note is closed.